sql/core/src/main/java/org/apache/spark/sql/sources/v2/SessionConfigSupport.java

```diff
@@ -27,10 +27,10 @@
 @InterfaceStability.Evolving
 public interface SessionConfigSupport extends DataSourceV2 {
```

Comment:
@cloud-fan, in the `TableCatalog` addition, #21306, I added this to configure catalogs. That way, a catalog is how you add configuration for all tables. Anonymous tables from the `ReadSupportProvider` classes don't really need this. I think we should remove the session config support in favor of using the catalog initialization config.

I also add a CaseInsensitiveStringMap for passing config that we should move to instead of DataSourceOptions. There are two main differences:
1. DataSourceOptions has methods to retrieve paths, table name, etc. that aren't needed when we use catalogs. Named tables and path-based tables should use catalogs and not pass these through options. Anonymous tables don't need standard ways to pass this information.
2. It is more clear what the class does: it provides a string to string mapping with case insensitive keys. It isn't as obvious what DataSourceOptions does.

Reply:
how about we keep `DataSourceOptions` a pure string-to-string map and move these `getPath`, `getTableName` methods to a wrapper class? Anyway we can do it in a follower up.

Reply:
If we are using `TableCatalog` and an equivalent `PathCatalog` then there is no need to pass these in options anyway. The table or path is passed directly to the table created by the catalog. It's fine for now, unless you want to get #21306 first and then update this to rely on the string-to-string map.